validator/remote/src/main/java/tech/pegasys/teku/validator/remote/typedef/handlers/SendSignedBlockRequest.java

```diff
@@ -13,28 +13,73 @@
 
 package tech.pegasys.teku.validator.remote.typedef.handlers;
 
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_NOT_ACCEPTABLE;
 import static tech.pegasys.teku.validator.remote.apiclient.ValidatorApiMethod.SEND_SIGNED_BLINDED_BLOCK;
 import static tech.pegasys.teku.validator.remote.apiclient.ValidatorApiMethod.SEND_SIGNED_BLOCK;
 
 import java.util.Collections;
+import java.util.Optional;
 import okhttp3.HttpUrl;
 import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
 import tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock;
 import tech.pegasys.teku.validator.api.SendSignedBlockResult;
 import tech.pegasys.teku.validator.remote.apiclient.ValidatorApiMethod;
 import tech.pegasys.teku.validator.remote.typedef.ResponseHandler;
 
 public class SendSignedBlockRequest extends AbstractTypeDefRequest {
 
-  public SendSignedBlockRequest(final HttpUrl baseEndpoint, final OkHttpClient okHttpClient) {
+  private final boolean preferSszBlockEncoding;
+  private boolean retryAsJson = false;
+
+  public SendSignedBlockRequest(
+      final HttpUrl baseEndpoint,
+      final OkHttpClient okHttpClient,
+      final boolean preferSszBlockEncoding) {
     super(baseEndpoint, okHttpClient);
+    this.preferSszBlockEncoding = preferSszBlockEncoding;
   }
 
   public SendSignedBlockResult sendSignedBlock(SignedBeaconBlock signedBeaconBlock) {
     final ValidatorApiMethod apiMethod =
         signedBeaconBlock.getMessage().getBody().isBlinded()
             ? SEND_SIGNED_BLINDED_BLOCK
             : SEND_SIGNED_BLOCK;
+
+    final SendSignedBlockResult result =
+        preferSszBlockEncoding
+            ? sendSignedBlockAsSszOrFallback(apiMethod, signedBeaconBlock)
+            : sendSignedBlockAsJson(apiMethod, signedBeaconBlock);
+    if (!result.isPublished() && retryAsJson) {
+      return sendSignedBlockAsJson(apiMethod, signedBeaconBlock);
+    }
+    return result;
+  }
+
+  public boolean isRetryAsJson() {
+    return retryAsJson;
```

Comment:
Probably worth renaming this and the variable - `retryAsJson` makes me think that it's a flag indicating if the request is allowed to retry as json if needed rather than a flag indicating it did need to retry.  Maybe something like `sszNotAcceptable`?